src/store/modules/imageUploaderStore.js

```diff
@@ -0,0 +1,64 @@
+import axios from 'axios';
+const aws = require('aws-sdk')
+
+
+export const imageUploaderStore = {
+    mutations: {
+        SET_IMAGEPATH(state, newState) {
+            this.state.imagePath = newState.data.imagePath
+        }
+    },
+
+    actions: {
+        upload_to_s3(context, payload) {
+            const file = payload;
+            aws.config.update({
+                secretAccessKey: process.env.DEV_SECRET_ACCESS_KEY,
+                accessKeyId: process.env.DEV_ACCESS_KEY_ID,
+            })
+            const s3 = new aws.S3({
+                signatureVersion: 'v4',
+                region: process.env.DEV_AWS_REGION,
+            })
+            
+
+            const key = 'static/uploads/' + Date.now() + '-' + file.name.replace(/\s/g, '-')
+            const params = {
+                Bucket: process.env.DEV_AWS_BUCKET,
+                Key: key,
+                Expires: 10,
+                ContentType: file.type,
+                
+            }
+            
+
+            const url = s3.getSignedUrl('putObject', params)
+            return axios
+                .put(url, file, {
+                    headers: {
+                        'Content-Type': file.type,
+                    },
+                })
+                .then(result => {
+                    const bucketUrl = decodeURIComponent(result.request.responseURL).split(
+                        key
+                    )[0]
+                    result.key = key
+                    result.fullPath = bucketUrl + key
```

Comment:
use a mutator for this. I'm very sure vue will scream at you for trying to do this inside an action. Fake the s3 upload if you are still having cors related issues and try what you have it, it wont work.